services/dynamic/dynamic-xml.service.js

```diff
@@ -41,21 +41,25 @@ module.exports = class DynamicXml extends BaseService {
       throw new InvalidParameter({ prettyMessage: e.message })
     }
 
-    if (!Array.isArray(values)) {
-      throw new InvalidResponse({
-        prettyMessage: 'unsupported query',
-      })
-    }
-
-    values = values.reduce((accum, node) => {
-      if (pathIsAttr) {
-        accum.push(node.value)
-      } else if (node.firstChild) {
-        accum.push(node.firstChild.data)
-      }
+    if (
+      typeof values === 'string' ||
+      typeof values === 'number' ||
+      typeof values === 'boolean'
+    ) {
+      values = [values]
+    } else {
+      values = values.reduce((accum, node) => {
```

Comment:
Are we sure that `xpath.select(...)` will always return a `string`, `number`, `boolean`, or an array? This seems like a superior fix for the type errors we were getting before, but I just want to make sure we can't still end up trying to call `.reduce` on an undefined `values` somehow

Reply:
@calebcartwright  I wasn't able to find a query which returns something different than types mentioned above. But we can revert previous code to support these unusual values returned by `select`.